Adding new Heading, Subheading, and Display text tokens

arniebradfo opened the pull request:
- Removed all references to Maple, Redwood, and Spruce option/tokens.
- Moved Maple, Redwood, and Spruce to deprecated file.
- Created a new font options file.
- Created new Heading, Subheading, and Display text tokens/mixins.
- Moved the old Header text tokens/mixins to deprecated file.

@KrisKnabel what do we do about the native `Sentinel-Semibold.otf` set of values in `mobile/text.json5`? should these be changed/updated to Stewart and Graphik?

@mhewson Are we loading the new REI Stuart and Graphik `@font-face` anywhere? Is that part of this story?

KrisKnabel commented:
> * Removed all references to Maple, Redwood, and Spruce option/tokens.
> * Moved Maple, Redwood, and Spruce to deprecated file.
> * Created a new font options file.
> * Created new Heading, Subheading, and Display text tokens/mixins.
> * Moved the old Header text tokens/mixins to deprecated file.
> 
> @KrisKnabel what do we do about the native `Sentinel-Semibold.otf` set of values in `mobile/text.json5`? should these be changed/updated to Stewart and Graphik?

If design is ready to rollout Stewart and Graphik for Native, than feel free to change them as long as I can get either a .otf or .ttf copy for each.

mhewson commented:
The font will not be hosted by us - its available on an internal CDN

KrisKnabel commented:
> The font will not be hosted by us - its available on an internal CDN

Not for Native Mobile, the fonts are pre-packaged before submitting the app to the app stores.

arniebradfo commented:
@KrisKnabel do those .otf/.ttf files live in this tokens repo?

KrisKnabel commented:
> @KrisKnabel do those .otf/.ttf files live in this tokens repo?

No they live in the Android and iOS Native Package repos.

arniebradfo commented:
@KrisKnabel Ok, I'll see if I can get the font files on Monday.

lancetharper commented:
This all looks ok to me, so go ahead and approve if all of the other discussion is resolved. I want to get this PR merged before I finish my changes for docs since there will be merge conflicts that I can take care of

lancetharper commented:
Do we need a changelog along with this too? We haven't made any breaking changes (or really any changes) to these tokens before so we've never actually had to do this yet.

@mhewson @arniebradfo

arniebradfo commented:
@lancetharper as far as a changelog goes, we did add tokens to the output so that should go in the changelog, but don't you only do a changelog for each release? I think we should wait until your deprecation work is done before making another alpha release.

lancetharper commented:
@arniebradfo We can keep a running changelog.md independent of releases so we actually know what's changed. It's just usually easier to document the changes as we make them rather than trying to remember everything at the end.

arniebradfo commented:
@lancetharper where is the `changelog.md`? is it not in the repo?

lancetharper commented:
> @lancetharper where is the `changelog.md`? is it not in the repo?

We don't have one since we haven't had to do this yet. We just need to add one at the root of the repo

arniebradfo commented:
@lancetharper I added a [Changelog.md](https://github.com/rei/rei-cedar-tokens/pull/54/commits/609f1147196c1aaf1099b71a8fe33fd40dc215c9#diff-9b8fab691c00b9e5380b19ce882f3271)

lancetharper commented on tokens/web/deprecated-2019-fall.json5:
```diff
@@ -0,0 +1,611 @@
+      // header-1
+      '1': {
+        family: {
+          value: '{options.font.family.serif.value}',
```
I just realized these are breaking changes for this, is that ok? 

Sure we're deprecating them, but this is also changing actual font families and stuff (like using Stuart instead of Roboto, etc.) so it isn't a backwards compatible deprecation...but are people going to be allowed to stay on the old font stack if they choose? How is that being handled?

arniebradfo replied:
@mhewson This was the plan, correct? All tokens get the updated font stack. I think it still is backwards compatible because the token names wont change, only their values. Changing a value doesn't qualify as a breaking change

arniebradfo commented:
@lancetharper & @mhewson 
I've reverted the old font header-tokens that are different to hardcoded values.
I've added `Sentinel` and `Roboto` to **All** font stacks. I think this will be the safest option. Let me know if you think otherwise...

@lancetharper here are the old to new token mappings:

Old | New
-- | --
`cdr-text-header-1` | `cdr-text-display-1200`
`cdr-text-header-2` | `cdr-text-display-1000`
`cdr-text-header-3` | `cdr-text-heading-800`
`cdr-text-header-4` | `cdr-text-heading-700`
`cdr-text-header-5` | `cdr-text-heading-600`
`cdr-text-header-6` | `cdr-text-heading-500`
`cdr-text-header-7` | `cdr-text-heading-400`
So you can add those to a deprecation message.

heres what it looks like visually:
![Screen Shot 2019-08-16 at 2 12 17 PM](https://user-images.githubusercontent.com/6248614/63199045-74fa4500-c031-11e9-9dc9-2b00f01bfa19.png)

arniebradfo commented:
@lancetharper here are the old to new token mappings:

Old | New
-- | --
`cdr-text-header-1` | `cdr-text-display-1200`
`cdr-text-header-2` | `cdr-text-display-1000`
`cdr-text-header-3` | `cdr-text-display-800`
`cdr-text-header-4` | `cdr-text-display-700`
`cdr-text-header-5` | `cdr-text-display-600`
`cdr-text-header-6` | `cdr-text-heading-500`
`cdr-text-header-7` | `cdr-text-heading-400`
So you can add those to a deprecation message.

heres what it looks like visually:
![Deprecation-Mappings-Opt-2](https://user-images.githubusercontent.com/6248614/63199378-8e4fc100-c032-11e9-99e5-340a52f44e47.png)

arniebradfo commented:
> 🎉 I'll put those mappings into the tokens on my deprecation branch. Thanks @arniebradfo

hold off just a minute jay might want to change them. Sorry I jumped the gun.

lancetharper commented:
> > 🎉 I'll put those mappings into the tokens on my deprecation branch. Thanks @arniebradfo
> 
> hold off just a minute jay might want to change them. Sorry I jumped the gun.

👌 That's why I always try to leave clicking the merge button to whoever opened the PR

arniebradfo commented:
Well, the mappings aren't in the PR. I don't know where they go.

arniebradfo commented:
Final Token Deprecation Replacement Recommendation Mappings:

Old | New
-- | --
`cdr-text-header-1` | `cdr-text-display-1100`
`cdr-text-header-2` | `cdr-text-display-900`
`cdr-text-header-3` | `cdr-text-heading-700`
`cdr-text-header-4` | `cdr-text-heading-600`
`cdr-text-header-5` | `cdr-text-heading-500`
`cdr-text-header-6` | `cdr-text-heading-400`
`cdr-text-header-7` | `cdr-text-heading-300`

![Screen Shot 2019-08-16 at 2 59 27 PM](https://user-images.githubusercontent.com/6248614/63200579-a6c1da80-c036-11e9-9ec7-526671610c23.png)
